cloudigrade/api/clouds/aws/tasks/inspection.py

```diff
@@ -2,578 +2,111 @@
-        same way for those other clouds.
-
-    Args:
-        cloud (str): identifier token for the cloud provider type (e.g. "aws")
-        ami_mountpoints (list): list of tuples each having (ami_id, mount_point)
-
-    Returns:
-        list of strings that make up the command arguments for the houndigrade task
-
-    """
-    task_command = ["-c", cloud]
-    for ami_id, mount_point in ami_mountpoints:
-        task_command.extend(["-t", ami_id, mount_point])
-    return task_command
-
-
-def _check_cluster_volume_mounts(ec2_instance_id, ami_mountpoints):
-    """
-    Check that the cluster instance has devices attached at the expected mount points.
-
-    Args:
-        ec2_instance_id (str): EC2 instance ID of the running houndigrade instance
-        ami_mountpoints (list): list of tuples each having (ami_id, mount_point)
-
-    Returns:
-        bool true if the cluster has devices attached at the expected mount points
-
-    """
-    instance = describe_cluster_instances([ec2_instance_id])[ec2_instance_id]
-    device_mappings = dict(
-        (mapping["DeviceName"], mapping) for mapping in instance["BlockDeviceMappings"]
-    )
-    incomplete_mounts = False
-    for ami_id, mount_point in ami_mountpoints:
-        device_mapping = device_mappings.get(mount_point, {})
-        status = device_mapping.get("Ebs", {}).get("Status")
-        if not device_mapping:
-            logger.warning(
-                _(
-                    "Expected device %(mount_point)s for %(ami_id)s not found in "
-                    "described instance %(ec2_instance_id)s"
-                ),
-                {
-                    "mount_point": mount_point,
-                    "ami_id": ami_id,
-                    "ec2_instance_id": ec2_instance_id,
-                },
-            )
-            incomplete_mounts = True
-        elif status == "attaching":
-            logger.info(
-                _(
-                    "Device %(mount_point)s for %(ami_id)s is still attaching to "
-                    "instance %(ec2_instance_id)s"
-                ),
-                {
-                    "mount_point": mount_point,
-                    "ami_id": ami_id,
-                    "ec2_instance_id": ec2_instance_id,
-                },
-            )
-            incomplete_mounts = True
-        elif status != "attached":
-            logger.error(
-                _(
-                    "Expected device %(mount_point)s for %(ami_id)s has unexpected "
-                    "status %(status)s for instance %(ec2_instance_id)s; "
-                    "current described block device mapping is %(detail)s"
-                ),
-                {
-                    "mount_point": mount_point,
-                    "ami_id": ami_id,
-                    "ec2_instance_id": ec2_instance_id,
-                    "status": status,
-                    "detail": device_mapping,
+                "DeviceName": "/dev/xvdbb",
```

Comment:
Since this is always going to be a brand new instance with exactly the number of attached devices we expect, this is probably fine, but I did pause when I saw the hard-coded path/name. Just putting a note here to double-check that expectation.

Reply:
Felt weird to me too, I considering defining it once at the top, but honestly this is just something I earmarked for change when/if we implement needed changes for LVM. At that point we will no longer need to designate a device name at all.